platforms/common/shutdown.cpp

```diff
@@ -36,6 +36,11 @@
  * Implementation of the API declared in px4_shutdown.h.
  */
 
+#ifdef __PX4_NUTTX
+// Make sure this is defined, this is a kernel module
+#define __KERNEL__
```

Comment:
why not just add something like "target_compile_options(mmymodule.... PRIVATE -D__KERNEL__)" into CMakeLists.txt for the module, which compiles this and goes into kernel? This is fine as well, don't really care :)

Reply:
Yes this is the preferred way but in this particular case, the library that takes shutdown.cpp is px4_platform and it is unambiguous where it goes. It is still not obvious / clear to me how that library should be handled. It is still just a massive mess in my head.

Reply:
I separated shutdown into its own target, now its compile time flags can be adjusted